Subject: [PATCH] change the query to get number as prisonNumber from the prisoner_prisoner table and amend the swagger docs
---
Index: src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/controller/ExternalMovementsController.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/controller/ExternalMovementsController.kt b/src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/controller/ExternalMovementsController.kt
--- a/src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/controller/ExternalMovementsController.kt	(revision dee9d6f6c1adab8b4d5a8cb5697a06a079394cf4)
+++ b/src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/controller/ExternalMovementsController.kt	(revision 3ef98572d3eb24dab9d856d202ee0f67861e29e5)
@@ -25,7 +25,7 @@
 
   @GetMapping("/external-movements/count")
   @Operation(
-    description = "Gets a count of external movements (mocked)",
+    description = "Gets a count of external movements",
     security = [ SecurityRequirement(name = "bearer-jwt") ],
   )
   fun count(
@@ -42,7 +42,7 @@
 
   @GetMapping("/external-movements")
   @Operation(
-    description = "Gets a list of external movements (mocked)",
+    description = "Gets a list of external movements",
     security = [ SecurityRequirement(name = "bearer-jwt") ],
   )
   fun externalMovements(
Index: src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/data/ExternalMovementPrisonerEntity.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/data/ExternalMovementPrisonerEntity.kt b/src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/data/ExternalMovementPrisonerEntity.kt
--- a/src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/data/ExternalMovementPrisonerEntity.kt	(revision dee9d6f6c1adab8b4d5a8cb5697a06a079394cf4)
+++ b/src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/data/ExternalMovementPrisonerEntity.kt	(revision 3ef98572d3eb24dab9d856d202ee0f67861e29e5)
@@ -4,7 +4,7 @@
 
 data class ExternalMovementPrisonerEntity(
   val id: Long,
-  val prisoner: Long,
+  val number: String,
   val firstName: String,
   val lastName: String,
   val date: LocalDateTime,
Index: src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/data/ExternalMovementRepositoryCustomImpl.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/data/ExternalMovementRepositoryCustomImpl.kt b/src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/data/ExternalMovementRepositoryCustomImpl.kt
--- a/src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/data/ExternalMovementRepositoryCustomImpl.kt	(revision dee9d6f6c1adab8b4d5a8cb5697a06a079394cf4)
+++ b/src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/data/ExternalMovementRepositoryCustomImpl.kt	(revision 3ef98572d3eb24dab9d856d202ee0f67861e29e5)
@@ -23,7 +23,7 @@
     val (preparedStatementNamedParams, whereClause) = constructWhereClause(filters)
     val sortingDirection = if (sortedAsc) "asc" else "desc"
 
-    val sql = """SELECT movements.id, movements.prisoner, prisoners.firstname, prisoners.lastname, movements.date, 
+    val sql = """SELECT movements.id, prisoners.number, prisoners.firstname, prisoners.lastname, movements.date, 
                     movements.time, to_char(movements.time, 'HH24:MI:SS') as timeOnly, movements.direction, movements.type, 
                     movements.origin, movements.destination, movements.reason 
                     FROM datamart.domain.movements_movements as movements
@@ -39,7 +39,7 @@
     ).map { q ->
       ExternalMovementPrisonerEntity(
         q["id"] as Long,
-        q["prisoner"] as Long,
+        q["number"] as String,
         q["firstname"] as String,
         q["lastname"] as String,
         (q["date"] as Timestamp).toLocalDateTime(),
Index: src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/model/ExternalMovementModel.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/model/ExternalMovementModel.kt b/src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/model/ExternalMovementModel.kt
--- a/src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/model/ExternalMovementModel.kt	(revision dee9d6f6c1adab8b4d5a8cb5697a06a079394cf4)
+++ b/src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/model/ExternalMovementModel.kt	(revision 3ef98572d3eb24dab9d856d202ee0f67861e29e5)
@@ -5,7 +5,6 @@
 
 data class ExternalMovementModel(
   val id: Long,
-  // This is the "prisoner" column in Redshift. Keeping is as prisonNumber for not breaking the UI for now and will change in the future.
   val prisonNumber: String,
   val firstName: String,
   val lastName: String,
Index: src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/service/ExternalMovementService.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/service/ExternalMovementService.kt b/src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/service/ExternalMovementService.kt
--- a/src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/service/ExternalMovementService.kt	(revision dee9d6f6c1adab8b4d5a8cb5697a06a079394cf4)
+++ b/src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/service/ExternalMovementService.kt	(revision 3ef98572d3eb24dab9d856d202ee0f67861e29e5)
@@ -21,7 +21,7 @@
 
   private fun toModel(entity: ExternalMovementPrisonerEntity): ExternalMovementModel {
     return ExternalMovementModel(
-      entity.id, entity.prisoner.toString(), entity.firstName, entity.lastName,
+      entity.id, entity.number, entity.firstName, entity.lastName,
       entity.date.toLocalDate(), entity.time.toLocalTime(), entity.origin, entity.destination, entity.direction, entity.type, entity.reason,
     )
   }
@@ -30,7 +30,7 @@
     return when (sortColumn) {
       "date" -> "date"
       "time" -> "timeOnly"
-      "prisonNumber" -> "prisoner"
+      "prisonNumber" -> "prisoners.number"
       "direction" -> "direction"
       "from" -> "origin"
       "to" -> "destination"
Index: src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/data/ExternalMovementRepositoryTest.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/data/ExternalMovementRepositoryTest.kt b/src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/data/ExternalMovementRepositoryTest.kt
--- a/src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/data/ExternalMovementRepositoryTest.kt	(revision dee9d6f6c1adab8b4d5a8cb5697a06a079394cf4)
+++ b/src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/data/ExternalMovementRepositoryTest.kt	(revision 3ef98572d3eb24dab9d856d202ee0f67861e29e5)
@@ -241,7 +241,7 @@
     val prisoner9846 = PrisonerEntity(9846, "W2505GF", "FirstName6", "LastName6", null)
     val movementPrisonerNullValues = ExternalMovementPrisonerEntity(
       6,
-      9846,
+      "W2505GF",
       "FirstName6",
       "LastName6",
       LocalDateTime.of(2050, 6, 1, 0, 0, 0),
@@ -370,7 +370,7 @@
   object AllMovementPrisoners {
     val allMovementPrisoners = allExternalMovements.mapIndexed { i, em ->
       ExternalMovementPrisonerEntity(
-        em.id, em.prisoner,
+        em.id, allPrisoners[i].number,
         "FirstName${i + 1}", "LastName${i + 1}", em.date, em.time,
         em.origin, em.destination, em.direction, em.type, em.reason,
       )
Index: src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/integration/ExternalMovementsIntegrationTest.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/integration/ExternalMovementsIntegrationTest.kt b/src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/integration/ExternalMovementsIntegrationTest.kt
--- a/src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/integration/ExternalMovementsIntegrationTest.kt	(revision dee9d6f6c1adab8b4d5a8cb5697a06a079394cf4)
+++ b/src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/integration/ExternalMovementsIntegrationTest.kt	(revision 3ef98572d3eb24dab9d856d202ee0f67861e29e5)
@@ -90,9 +90,9 @@
       .expectBody()
       .json(
         """[
-        {"id": 5, "prisonNumber":"6851","firstName": "${prisoner6851.firstName}", "lastName": "${prisoner6851.lastName}","date":"2023-05-20","time":"14:00:00","from":"Isle of Wight","to":"Northumberland","direction":"In","type":"Transfer","reason":"Transfer In from Other Establishment"},
-        {"id": 4, "prisonNumber":"7849","firstName": "${prisoner7849.firstName}", "lastName": "${prisoner7849.lastName}","date":"2023-05-01","time":"15:19:00","from":"Cardiff","to":"Maidstone","direction":"Out","type":"Transfer","reason":"Transfer Out to Other Establishment"},
-        {"id": 3, "prisonNumber":"4800","firstName": "${prisoner4800.firstName}", "lastName": "${prisoner4800.lastName}","date":"2023-04-30","time":"13:19:00","from":"Wakefield","to":"Dartmoor","direction":"In","type":"Transfer","reason":"Transfer In from Other Establishment"}
+        {"id": 5, "prisonNumber":"${prisoner6851.number}","firstName": "${prisoner6851.firstName}", "lastName": "${prisoner6851.lastName}","date":"2023-05-20","time":"14:00:00","from":"Isle of Wight","to":"Northumberland","direction":"In","type":"Transfer","reason":"Transfer In from Other Establishment"},
+        {"id": 4, "prisonNumber":"${prisoner7849.number}","firstName": "${prisoner7849.firstName}", "lastName": "${prisoner7849.lastName}","date":"2023-05-01","time":"15:19:00","from":"Cardiff","to":"Maidstone","direction":"Out","type":"Transfer","reason":"Transfer Out to Other Establishment"},
+        {"id": 3, "prisonNumber":"${prisoner4800.number}","firstName": "${prisoner4800.firstName}", "lastName": "${prisoner4800.lastName}","date":"2023-04-30","time":"13:19:00","from":"Wakefield","to":"Dartmoor","direction":"In","type":"Transfer","reason":"Transfer In from Other Establishment"}
       ]       
       """,
       )
@@ -114,11 +114,11 @@
       .json(
         """
       [
-        {"id": 5, "prisonNumber":"6851", "firstName": "${prisoner6851.firstName}", "lastName": "${prisoner6851.lastName}", "date":"2023-05-20","time":"14:00:00","from":"Isle of Wight","to":"Northumberland","direction":"In","type":"Transfer","reason":"Transfer In from Other Establishment"},
-        {"id": 4, "prisonNumber":"7849", "firstName": "${prisoner7849.firstName}", "lastName": "${prisoner7849.lastName}", "date":"2023-05-01","time":"15:19:00","from":"Cardiff","to":"Maidstone","direction":"Out","type":"Transfer","reason":"Transfer Out to Other Establishment"},
-        {"id": 3, "prisonNumber":"4800", "firstName": "${prisoner4800.firstName}", "lastName": "${prisoner4800.lastName}", "date":"2023-04-30","time":"13:19:00","from":"Wakefield","to":"Dartmoor","direction":"In","type":"Transfer","reason":"Transfer In from Other Establishment"},
-        {"id": 2, "prisonNumber":"5207", "firstName": "${prisoner5207.firstName}", "lastName": "${prisoner5207.lastName}", "date": "2023-04-25","time":"12:19:00","from":"Elmley","to":"Pentonville","direction":"In","type":"Transfer","reason":"Transfer In from Other Establishment"},
-        {"id": 1, "prisonNumber":"8894", "firstName": "${prisoner8894.firstName}", "lastName": "${prisoner8894.lastName}", "date": "2023-01-31","time":"03:01:00","from":"Ranby","to":"Kirkham","direction":"In","type":"Admission","reason":"Unconvicted Remand"}
+        {"id": 5, "prisonNumber":"${prisoner6851.number}", "firstName": "${prisoner6851.firstName}", "lastName": "${prisoner6851.lastName}", "date":"2023-05-20","time":"14:00:00","from":"Isle of Wight","to":"Northumberland","direction":"In","type":"Transfer","reason":"Transfer In from Other Establishment"},
+        {"id": 4, "prisonNumber":"${prisoner7849.number}", "firstName": "${prisoner7849.firstName}", "lastName": "${prisoner7849.lastName}", "date":"2023-05-01","time":"15:19:00","from":"Cardiff","to":"Maidstone","direction":"Out","type":"Transfer","reason":"Transfer Out to Other Establishment"},
+        {"id": 3, "prisonNumber":"${prisoner4800.number}", "firstName": "${prisoner4800.firstName}", "lastName": "${prisoner4800.lastName}", "date":"2023-04-30","time":"13:19:00","from":"Wakefield","to":"Dartmoor","direction":"In","type":"Transfer","reason":"Transfer In from Other Establishment"},
+        {"id": 2, "prisonNumber":"${prisoner5207.number}", "firstName": "${prisoner5207.firstName}", "lastName": "${prisoner5207.lastName}", "date": "2023-04-25","time":"12:19:00","from":"Elmley","to":"Pentonville","direction":"In","type":"Transfer","reason":"Transfer In from Other Establishment"},
+        {"id": 1, "prisonNumber":"${prisoner8894.number}", "firstName": "${prisoner8894.firstName}", "lastName": "${prisoner8894.lastName}", "date": "2023-01-31","time":"03:01:00","from":"Ranby","to":"Kirkham","direction":"In","type":"Admission","reason":"Unconvicted Remand"}
       ]
       """,
       )
@@ -217,7 +217,7 @@
       .expectBody()
       .json(
         """[
-         {"id": 4, "prisonNumber": "7849", "firstName": "${prisoner7849.firstName}", "lastName": "${prisoner7849.lastName}", "date": "2023-05-01", "time": "15:19:00", "from": "Cardiff", "to": "Maidstone", "direction": "Out", "type": "Transfer", "reason": "Transfer Out to Other Establishment"}
+         {"id": 4, "prisonNumber": "${prisoner7849.number}", "firstName": "${prisoner7849.firstName}", "lastName": "${prisoner7849.lastName}", "date": "2023-05-01", "time": "15:19:00", "from": "Cardiff", "to": "Maidstone", "direction": "Out", "type": "Transfer", "reason": "Transfer Out to Other Establishment"}
       ]       
       """,
       )
Index: src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/service/ExternalMovementServiceTest.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/service/ExternalMovementServiceTest.kt b/src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/service/ExternalMovementServiceTest.kt
--- a/src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/service/ExternalMovementServiceTest.kt	(revision dee9d6f6c1adab8b4d5a8cb5697a06a079394cf4)
+++ b/src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/service/ExternalMovementServiceTest.kt	(revision 3ef98572d3eb24dab9d856d202ee0f67861e29e5)
@@ -47,7 +47,7 @@
   @CsvSource(
     "date, date",
     "time, timeOnly",
-    "prisonNumber, prisoner",
+    "prisonNumber, prisoners.number",
     "direction, direction",
     "from, origin",
     "to, destination",
@@ -68,7 +68,7 @@
   object AllModels {
     val externalMovement1 = ExternalMovementModel(
       1,
-      "8894",
+      AllEntities.externalMovement1.number,
       prisoner8894.firstName,
       prisoner8894.lastName,
       LocalDate.of(2023, 1, 31),
@@ -81,7 +81,7 @@
     )
     val externalMovement2 = ExternalMovementModel(
       2,
-      "5207",
+      AllEntities.externalMovement2.number,
       prisoner5207.firstName,
       prisoner5207.lastName,
       LocalDate.of(2023, 4, 25),
@@ -101,7 +101,7 @@
   object AllEntities {
     val externalMovement1 = ExternalMovementPrisonerEntity(
       1,
-      prisoner8894.id,
+      prisoner8894.number,
       prisoner8894.firstName,
       prisoner8894.lastName,
       LocalDateTime.of(2023, 1, 31, 0, 0, 0),
@@ -114,7 +114,7 @@
     )
     val externalMovement2 = ExternalMovementPrisonerEntity(
       2,
-      prisoner5207.id,
+      prisoner5207.number,
       prisoner5207.firstName,
       prisoner5207.lastName,
       LocalDateTime.of(2023, 4, 25, 0, 0, 0),
